File: iOS/Sample/Sample/Http/ApiRequester.swift

import Foundation

class ApiRequester {
    
    // APIをコールする
    class func post(params: [String: String],
                    completion: @escaping ((Bool, Any?) -> ())) {
        
        // HTTP通信を行う
        HttpRequester.post(url: Constants.ServerApiUrl, params: params) { result, data in
            
            // HTTP通信に失敗
            guard result, let data = data else {
                completion(false, nil)
                return
            }
            do {
                // JSONに変換
                let options = JSONSerialization.ReadingOptions.allowFragments
                let json = try JSONSerialization.jsonObject(with: data,
                                                            options: options)
                let dic = json as! Dictionary<String, Any>
                
                // 呼び出し元に通知
                completion(true, dic)
                
            } catch {
                // JSON変換に失敗
                completion(false, nil)
            }
        }
    }
}

File: Android/app/src/main/java/matching/sample/Http/ApiRequester.java
package matching.sample.Http;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.HashMap;
import java.util.StringJoiner;

import matching.sample.System.Constants;

public class ApiRequester {

    // APIをコール
    public static void post(HashMap<String, String> params,
                            final Callback callback) {

        HttpRequester.RequestType requestType = HttpRequester.RequestType.string;

        HttpRequester http = new HttpRequester(requestType,
                new HttpRequester.Callback() {

            @Override
            public void didReceiveData(boolean result, Object data) {
                // 成功
                if (result) {
                    try {
                        // JSON変換
                        JSONObject json = new JSONObject((String)data);

                        // 呼び出し元に成功を通知
                        callback.didReceiveData(true, json);
                        return;

                    } catch (JSONException e) { }
                }

                // 呼び出し元に失敗を通知
                callback.didReceiveData(false, null);
            }
        });

        // POSTパラメータを格納する
        StringJoiner paramJoiner = new StringJoiner("&");
        for (String key : params.keySet()) {
            paramJoiner.add(key + "=" + params.get(key));
        }

        // HTTP通信を実行
        http.execute(Constants.ServerApiUrl, "POST", paramJoiner.toString());
    }

    // コールバック
    public interface Callback {
        void didReceiveData(boolean result, JSONObject json);
    }
}